src/azure-cli/azure/cli/command_modules/appconfig/_kv_export_helpers.py

```diff
@@ -0,0 +1,441 @@
+        return
+
+    try:
+        non_slot_settings = []
+        slot_settings = []
+        for kv in key_values:
+            name = kv.key
+            value = kv.value
+            # If its a KeyVault ref, convert the format to AppService KeyVault ref format
+            if __is_key_vault_ref(kv):
+                try:
+                    secret_uri = json.loads(value).get("uri")
+                    if secret_uri:
+                        value = (
+                            AppServiceConstants.APPSVC_KEYVAULT_PREFIX +
+                            "(SecretUri={0})".format(secret_uri)
+                        )
+                    else:
+                        logger.debug(
+                            'Key "%s" with value "%s" is not a well-formatted KeyVault reference.It will be treated like a regular key-value.\n',
+                            name,
+                            value,
+                        )
+                except (AttributeError, TypeError, ValueError) as e:
+                    logger.debug(
+                        'Key "%s" with value "%s" is not a well-formatted KeyVault reference. It will be treated like a regular key-value.\n%s',
+                        name,
+                        value,
+                        str(e),
+                    )
+
+            if (
+                AppServiceConstants.APPSVC_SLOT_SETTING_KEY in kv.tags and
+                kv.tags[AppServiceConstants.APPSVC_SLOT_SETTING_KEY] == "true"
+            ):
+                slot_settings.append(name + "=" + value)
+            else:
+                non_slot_settings.append(name + "=" + value)
+        # known issue 4/26: with in-place update, AppService could change slot-setting true/false incorrectly
+        slot = (
+            appservice_account.get("resource_name")
+            if appservice_account.get("resource_type") == "slots"
+            else None
+        )
+        from azure.cli.command_modules.appservice.custom import update_app_settings
+
+        update_app_settings(
+            cmd,
+            resource_group_name=appservice_account["resource_group"],
+            name=appservice_account["name"],
+            settings=non_slot_settings,
+            slot_settings=slot_settings,
+            slot=slot,
+        )
+    except Exception as exception:
+        raise CLIError("Failed to write key-values to appservice: " + str(exception))
+
+
+def __write_kv_and_features_to_file(
+    file_path,
+    key_values=None,
+    features=None,
+    format_=None,
+    separator=None,
+    skip_features=False,
+    naming_convention="pascal",
+):
+    if not key_values and not features:
+        logger.warning("\nSource configuration is empty. No changes will be made.")
+        return
+
+    try:
+        exported_keyvalues = __export_keyvalues(key_values, format_, separator, None)
+        if features and not skip_features:
+            compatibility_mode = os.environ.get(
```

Comment:
Did we consider using `az configure --defaults` instead of environment variable for this?

Reply:
Yes, we saw it was better to have the `compatibility mode` as an environment variable instead of an argument because its  much less "sticky" we could easily remove it compared to having an argument once users move to using the new fm schema completely.

Reply:
I dont think `az configure --defaults` can be considered an argument. But I agree that its a more permanent solution than the short term environment variable option.